### PR TITLE
vtk: clean up / reduce code repetition

### PR DESCRIPTION
**Pull request by lheckemann**
###### Motivation for this change
This refactors the vtk expressions to use a generic base expression,
which is used for all 3 versions.

vtk 7.x no longer uses gcc 8, since the quirk of it not building with
gcc 9 seems to have been fixed.

This also makes the python bindings available for all 3 versions, and
fixes building them for vtk 8 by adding a patch.

cc @tfmoraes @knedlsepp I've listed you as maintainers for all 3 versions, hope that's alright — if not we can of course change that back. I've also added myself.

###### Things done

- [x] Built all combinations of (pkgs, python3Packages, python2Packages).(vtk_7, vtk_8, vtk_9)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/vtk/generic.nix
Comment (FRidh) [inline pkgs/development/libraries/vtk/generic.nix:14]:
```
pythonMajor = lib.strings.substring 0 1 pythonInterpreter.pythonVersion;
```

File: pkgs/development/libraries/vtk/generic.nix
Comment (FRidh) [inline pkgs/development/libraries/vtk/generic.nix:16]: What's the advantage of moving `toPythonModule` and thereby deviating from the common approach? I think its important that contributors to `python-packages.nix` see they need to use `toPythonModule` in case of bindings, to reduce the chance in forgetting to add that.

Comment (lheckemann) [reply pkgs/development/libraries/vtk/generic.nix]: Reducing the amount of repetition. Not a hill I'm going to die on though, undid this change :)

**Comment by lheckemann** [comment]
@ofborg build vtk_7 vtk_8 vtk_9 python2Packages.vtk_7 python2Packages.vtk_8 python2Packages.vtk_9 python3Packages.vtk_7 python3Packages.vtk_8 python3Packages.vtk_9

**Review by prusnak — CHANGES_REQUESTED** [review]
It seems counterintuitive that `python3Packages.vtk` points to `python3Packages.vtk_7` while `pkgs.vtk` points to `pkgs.vtk_8`. Is there a good reason for that?

**Comment by lheckemann** [comment]
@prusnak only that that's no change from before (which in turn might be because vtk 8 didn't build with python 3 prior to this PR). I do want to clean that up and see if I can update all the dependents to vtk 9 once this is merged.

**Comment by knedlsepp** [comment]
Thanks for the cleanup and marking myself as maintainer on the remaining versions.

**Comment by tfmoraes** [comment]
@lheckemann good idea. Thanks for the modifications!

**Review by prusnak — APPROVED** [review]

**Pull request merged**
